mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -138,6 +140,82 @@ template <typename T> struct MILARewritePattern : public OpRewritePattern<T> {
     return nAlloc->getResult(0);
   }
 
+  Value createCollapseShapeOp(PatternRewriter &b, Location loc,
+                              Value source) const {
+    auto ctx = b.getContext();
+    auto sourceType = source.getType().cast<ShapedType>();
+    assert(sourceType.hasStaticShape() &&
+           "Only memrefs with static shapes are allowed");
+
+    auto shape = sourceType.getShape();
+    uint64_t collapsedDim = 1;
+    SmallVector<AffineExpr, 2> exprs;
+    for (uint32_t dim = 0; dim < shape.size(); ++dim) {
+      collapsedDim *= shape[dim];
+      exprs.push_back(getAffineDimExpr(dim, ctx));
+    }
+
+    SmallVector<int64_t, 1> collapsedShape;
+    SmallVector<ReassociationExprs, 1> reassocs;
+    collapsedShape.push_back(collapsedDim);
+    reassocs.push_back(exprs);
+
+    auto collapsedType =
+        MemRefType::get(collapsedShape, sourceType.getElementType(), {}, 5);
+    Value result =
+        b.create<memref::CollapseShapeOp>(loc, collapsedType, source, reassocs);
+    return result;
+  }
```

Comment:
Actually I tried but wasn't able to find which one to use. Can you point me any example? this is simply flattening to 1d and all the other dimensions are dummies. for example [1,1,1,1,1,16] ->[16]

Reply:
You'll want to `addDim()` in the dummy dimensions and `PassThrough` the 16

Reply:
Wouldn't ### merge be better

Reply:
merge works but it's actually ignoring the dimensions. I still can't find existing set of functions. I though top down ct builder should be used for this case but passThrough() is missing. Maybe I need to call addTransform?

Reply:
After have a talk with @sjw36, this is only feeding the result into linalg so better to keep as it is so that we can avoid unnecessary time for lowering/optimizing dummy dimensions.